### ISSUE TITLE
InvalidArgumentException: "Cannot send empty batch" (EXCEPTION) in "src/pocketmine/Server" at line 1884

### ISSUE THREAD

**Issue by teracube** [JackMD/MysteryCrate#59]
MysteryCrate_dev-72.phar
PocketMine-MP version 1.7dev-1079 "[REDACTED]" (API 3.0.0-ALPHA12)

```
2018-06-02 [17:17:58] [Server thread/CRITICAL]: Could not pass event 'pocketmine\event\entity\EntityLevelChangeEvent' to 'MysteryCrate v6.0.0': Cannot send empty batch on JackMD\MysteryCrate\EventListener
2018-06-02 [17:17:58] [Server thread/CRITICAL]: InvalidArgumentException: "Cannot send empty batch" (EXCEPTION) in "src/pocketmine/Server" at line 1884
2018-06-02 [17:17:58] [Server thread/DEBUG]: #0 src/pocketmine/level/Level(467): pocketmine\Server->batchPackets(array Array(), array Array(), boolean )
2018-06-02 [17:17:58] [Server thread/DEBUG]: #1 MysteryCrate_dev-72.phar/src/JackMD/MysteryCrate/EventListener(309): pocketmine\level\Level->addParticle(pocketmine\level\particle\FloatingTextParticle object, array Array())
2018-06-02 [17:17:58] [Server thread/DEBUG]: #2 src/pocketmine/plugin/MethodEventExecutor(38): JackMD\MysteryCrate\EventListener->onLevelChange(pocketmine\event\entity\EntityLevelChangeEvent object)
2018-06-02 [17:17:58] [Server thread/DEBUG]: #3 src/pocketmine/plugin/RegisteredListener(98): pocketmine\plugin\MethodEventExecutor->execute(JackMD\MysteryCrate\EventListener object, pocketmine\event\entity\EntityLevelChangeEvent object)
2018-06-02 [17:17:58] [Server thread/DEBUG]: #4 src/pocketmine/plugin/PluginManager(701): pocketmine\plugin\RegisteredListener->callEvent(pocketmine\event\entity\EntityLevelChangeEvent object)
2018-06-02 [17:17:58] [Server thread/DEBUG]: #5 src/pocketmine/entity/Entity(1878): pocketmine\plugin\PluginManager->callEvent(pocketmine\event\entity\EntityLevelChangeEvent object)
2018-06-02 [17:17:58] [Server thread/DEBUG]: #6 src/pocketmine/Player(925): pocketmine\entity\Entity->switchLevel(pocketmine\level\Level object)
2018-06-02 [17:17:58] [Server thread/DEBUG]: #7 src/pocketmine/entity/Entity(1742): pocketmine\Player->switchLevel(pocketmine\level\Level object)
2018-06-02 [17:17:58] [Server thread/DEBUG]: #8 src/pocketmine/entity/Entity(1767): pocketmine\entity\Entity->setPosition(pocketmine\level\Position object)
2018-06-02 [17:17:58] [Server thread/DEBUG]: #9 src/pocketmine/entity/Entity(1860): pocketmine\entity\Entity->setPositionAndRotation(pocketmine\level\Position object, double 51.251312255859, double 25.345901489258)
2018-06-02 [17:17:58] [Server thread/DEBUG]: #10 src/pocketmine/Player(3746): pocketmine\entity\Entity->teleport(pocketmine\level\Position object, NULL , NULL )
2018-06-02 [17:17:58] [Server thread/DEBUG]: #11 essentialsTP_dev-30.phar/src/essentialsTP/essentialsTP(563): pocketmine\Player->teleport(pocketmine\level\Position object)
2018-06-02 [17:17:58] [Server thread/DEBUG]: #12 src/pocketmine/command/PluginCommand(58): essentialsTP\essentialsTP->onCommand(pocketmine\Player object, pocketmine\command\PluginCommand object, string home, array Array())
2018-06-02 [17:17:58] [Server thread/DEBUG]: #13 src/pocketmine/command/SimpleCommandMap(258): pocketmine\command\PluginCommand->execute(pocketmine\Player object, string home, array Array())
2018-06-02 [17:17:58] [Server thread/DEBUG]: #14 src/pocketmine/Server(1978): pocketmine\command\SimpleCommandMap->dispatch(pocketmine\Player object, string home base)
2018-06-02 [17:17:58] [Server thread/DEBUG]: #15 src/pocketmine/Player(2191): pocketmine\Server->dispatchCommand(pocketmine\Player object, string home base)
2018-06-02 [17:17:58] [Server thread/DEBUG]: #16 src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(222): pocketmine\Player->chat(string /home base)
2018-06-02 [17:17:58] [Server thread/DEBUG]: #17 src/pocketmine/network/mcpe/protocol/CommandRequestPacket(54): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleCommandRequest(pocketmine\network\mcpe\protocol\CommandRequestPacket object)
2018-06-02 [17:17:58] [Server thread/DEBUG]: #18 src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(92): pocketmine\network\mcpe\protocol\CommandRequestPacket->handle(pocketmine\network\mcpe\PlayerNetworkSessionAdapter object)
2018-06-02 [17:17:58] [Server thread/DEBUG]: #19 src/pocketmine/network/mcpe/protocol/BatchPacket(114): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleDataPacket(pocketmine\network\mcpe\protocol\CommandRequestPacket object)
2018-06-02 [17:17:58] [Server thread/DEBUG]: #20 src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(92): pocketmine\network\mcpe\protocol\BatchPacket->handle(pocketmine\network\mcpe\PlayerNetworkSessionAdapter object)
2018-06-02 [17:17:58] [Server thread/DEBUG]: #21 src/pocketmine/Player(3033): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleDataPacket(pocketmine\network\mcpe\protocol\BatchPacket object)
2018-06-02 [17:17:58] [Server thread/DEBUG]: #22 src/pocketmine/network/mcpe/RakLibInterface(158): pocketmine\Player->handleDataPacket(pocketmine\network\mcpe\protocol\BatchPacket object)
2018-06-02 [17:17:58] [Server thread/DEBUG]: #23 vendor/pocketmine/raklib/server/ServerHandler(98): pocketmine\network\mcpe\RakLibInterface->handleEncapsulated(string 88.163.159.23 55555, raklib\protocol\EncapsulatedPacket object, integer 0)
2018-06-02 [17:17:58] [Server thread/DEBUG]: #24 src/pocketmine/network/mcpe/RakLibInterface(103): raklib\server\ServerHandler->handlePacket()
2018-06-02 [17:17:58] [Server thread/DEBUG]: #25 src/pocketmine/network/Network(94): pocketmine\network\mcpe\RakLibInterface->process()
2018-06-02 [17:17:58] [Server thread/DEBUG]: #26 src/pocketmine/network/mcpe/RakLibInterface(80): pocketmine\network\Network->processInterface(pocketmine\network\mcpe\RakLibInterface object)
2018-06-02 [17:17:58] [Server thread/DEBUG]: #27 vendor/pocketmine/snooze/src/SleeperHandler(113): pocketmine\network\mcpe\RakLibInterface->pocketmine\network\mcpe\{closure}()
2018-06-02 [17:17:58] [Server thread/DEBUG]: #28 vendor/pocketmine/snooze/src/SleeperHandler(82): pocketmine\snooze\SleeperHandler->processNotifications()
2018-06-02 [17:17:58] [Server thread/DEBUG]: #29 src/pocketmine/Server(2279): pocketmine\snooze\SleeperHandler->sleepUntil(double 1527952678.6945)
2018-06-02 [17:17:58] [Server thread/DEBUG]: #30 src/pocketmine/Server(2150): pocketmine\Server->tickProcessor()
2018-06-02 [17:17:58] [Server thread/DEBUG]: #31 src/pocketmine/Server(1733): pocketmine\Server->start()
2018-06-02 [17:17:58] [Server thread/DEBUG]: #32 src/pocketmine/PocketMine(290): pocketmine\Server->__construct(BaseClassLoader object, pocketmine\utils\MainLogger object, string /home/g3.hmserv.eu-19144/, string /home/g3.hmserv.eu-19144/plugins/)
2018-06-02 [17:17:58] [Server thread/DEBUG]: #33 /home/g3.hmserv.eu-19144/PocketMine-MP.phar(1): require(string phar:///home/g3.hmserv.eu-19144/PocketMine-MP.phar/src/pocketmine/PocketMine.php)
```

**Comment by ranski** [comment]
having the same problem :(

**Comment by JackMD** [comment]
This error is not occurring to me. Please give **Steps to Reproduce**

**Comment by teracube** [comment]
@JackMD 
sorry i don' t know

**Comment by JackMD** [comment]
@teracube and @ranski I need steps to reproduce this issue. Without it I can say/do nothing. It does not happen to me. Also does it happen all the time or just at random? Other than that I'm closing the issue until it occurs to me or I am pfovided with steps to reproduce.

**Comment by ranski** [comment]
@JackMD is there any way that i can send my server files so that you can reproduce the error?

**Comment by JackMD** [comment]
@ranski does the error occur to you at all times or at random. Also paste your `config.yml` file and please add ``` to he start and the end of the file.

**Comment by ranski** [comment]
@JackMD  Nvm , it seems to have been fixed, i installed the latest pmmp pocketmine from jenkins, (the last successfull build) and re installed the plugin from its source code

**Comment by JackMD** [comment]
Okay

**Comment by ranski** [comment]
@teracube you should also try that, it seems to fix it. Good luck!

**Comment by teracube** [comment]
@JackMD 
Ooops not fixed for me
today same issue

```
2018-06-14 [18:04:02] [Server thread/CRITICAL]: Could not pass event 'pocketmine\event\entity\EntityLevelChangeEvent' to 'MysteryCrate v6.0.0': Cannot send empty batch on JackMD\MysteryCrate\EventListener
2018-06-14 [18:04:02] [Server thread/CRITICAL]: InvalidArgumentException: "Cannot send empty batch" (EXCEPTION) in "src/pocketmine/Server" at line 1884
2018-06-14 [18:04:02] [Server thread/DEBUG]: #0 src/pocketmine/level/Level(467): pocketmine\Server->batchPackets(array Array(), array Array(), boolean )
2018-06-14 [18:04:02] [Server thread/DEBUG]: #1 MysteryCrate_dev-72.phar/src/JackMD/MysteryCrate/EventListener(309): pocketmine\level\Level->addParticle(pocketmine\level\particle\FloatingTextParticle object, array Array())
2018-06-14 [18:04:02] [Server thread/DEBUG]: #2 src/pocketmine/plugin/MethodEventExecutor(38): JackMD\MysteryCrate\EventListener->onLevelChange(pocketmine\event\entity\EntityLevelChangeEvent object)
2018-06-14 [18:04:02] [Server thread/DEBUG]: #3 src/pocketmine/plugin/RegisteredListener(98): pocketmine\plugin\MethodEventExecutor->execute(JackMD\MysteryCrate\EventListener object, pocketmine\event\entity\EntityLevelChangeEvent object)
2018-06-14 [18:04:02] [Server thread/DEBUG]: #4 src/pocketmine/plugin/PluginManager(701): pocketmine\plugin\RegisteredListener->callEvent(pocketmine\event\entity\EntityLevelChangeEvent object)
2018-06-14 [18:04:02] [Server thread/DEBUG]: #5 src/pocketmine/entity/Entity(1878): pocketmine\plugin\PluginManager->callEvent(pocketmine\event\entity\EntityLevelChangeEvent object)
2018-06-14 [18:04:02] [Server thread/DEBUG]: #6 src/pocketmine/Player(925): pocketmine\entity\Entity->switchLevel(pocketmine\level\Level object)
2018-06-14 [18:04:02] [Server thread/DEBUG]: #7 src/pocketmine/entity/Entity(1742): pocketmine\Player->switchLevel(pocketmine\level\Level object)
2018-06-14 [18:04:02] [Server thread/DEBUG]: #8 src/pocketmine/entity/Entity(1767): pocketmine\entity\Entity->setPosition(pocketmine\level\Position object)
2018-06-14 [18:04:02] [Server thread/DEBUG]: #9 src/pocketmine/entity/Entity(1860): pocketmine\entity\Entity->setPositionAndRotation(pocketmine\level\Position object, double 247.29554748535, double 16.097091674805)
2018-06-14 [18:04:02] [Server thread/DEBUG]: #10 src/pocketmine/Player(3746): pocketmine\entity\Entity->teleport(pocketmine\level\Position object, NULL , NULL )
2018-06-14 [18:04:02] [Server thread/DEBUG]: #11 essentialsTP_dev-30.phar/src/essentialsTP/essentialsTP(563): pocketmine\Player->teleport(pocketmine\level\Position object)
2018-06-14 [18:04:02] [Server thread/DEBUG]: #12 src/pocketmine/command/PluginCommand(58): essentialsTP\essentialsTP->onCommand(pocketmine\Player object, pocketmine\command\PluginCommand object, string home, array Array())
2018-06-14 [18:04:02] [Server thread/DEBUG]: #13 src/pocketmine/command/SimpleCommandMap(258): pocketmine\command\PluginCommand->execute(pocketmine\Player object, string home, array Array())
2018-06-14 [18:04:02] [Server thread/DEBUG]: #14 src/pocketmine/Server(1978): pocketmine\command\SimpleCommandMap->dispatch(pocketmine\Player object, string home base)
2018-06-14 [18:04:02] [Server thread/DEBUG]: #15 src/pocketmine/Player(2191): pocketmine\Server->dispatchCommand(pocketmine\Player object, string home base)
2018-06-14 [18:04:02] [Server thread/DEBUG]: #16 src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(222): pocketmine\Player->chat(string /home base)
2018-06-14 [18:04:02] [Server thread/DEBUG]: #17 src/pocketmine/network/mcpe/protocol/CommandRequestPacket(54): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleCommandRequest(pocketmine\network\mcpe\protocol\CommandRequestPacket object)
2018-06-14 [18:04:02] [Server thread/DEBUG]: #18 src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(92): pocketmine\network\mcpe\protocol\CommandRequestPacket->handle(pocketmine\network\mcpe\PlayerNetworkSessionAdapter object)
2018-06-14 [18:04:02] [Server thread/DEBUG]: #19 src/pocketmine/network/mcpe/protocol/BatchPacket(114): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleDataPacket(pocketmine\network\mcpe\protocol\CommandRequestPacket object)
2018-06-14 [18:04:02] [Server thread/DEBUG]: #20 src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(92): pocketmine\network\mcpe\protocol\BatchPacket->handle(pocketmine\network\mcpe\PlayerNetworkSessionAdapter object)
2018-06-14 [18:04:02] [Server thread/DEBUG]: #21 src/pocketmine/Player(3033): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleDataPacket(pocketmine\network\mcpe\protocol\BatchPacket object)
2018-06-14 [18:04:02] [Server thread/DEBUG]: #22 src/pocketmine/network/mcpe/RakLibInterface(158): pocketmine\Player->handleDataPacket(pocketmine\network\mcpe\protocol\BatchPacket object)
2018-06-14 [18:04:02] [Server thread/DEBUG]: #23 vendor/pocketmine/raklib/server/ServerHandler(98): pocketmine\network\mcpe\RakLibInterface->handleEncapsulated(string 176.155.161.200 40973, raklib\protocol\EncapsulatedPacket object, integer 0)
2018-06-14 [18:04:02] [Server thread/DEBUG]: #24 src/pocketmine/network/mcpe/RakLibInterface(103): raklib\server\ServerHandler->handlePacket()
2018-06-14 [18:04:02] [Server thread/DEBUG]: #25 src/pocketmine/network/Network(94): pocketmine\network\mcpe\RakLibInterface->process()
2018-06-14 [18:04:02] [Server thread/DEBUG]: #26 src/pocketmine/network/mcpe/RakLibInterface(80): pocketmine\network\Network->processInterface(pocketmine\network\mcpe\RakLibInterface object)
2018-06-14 [18:04:02] [Server thread/DEBUG]: #27 vendor/pocketmine/snooze/src/SleeperHandler(113): pocketmine\network\mcpe\RakLibInterface->pocketmine\network\mcpe\{closure}()
2018-06-14 [18:04:02] [Server thread/DEBUG]: #28 vendor/pocketmine/snooze/src/SleeperHandler(82): pocketmine\snooze\SleeperHandler->processNotifications()
2018-06-14 [18:04:02] [Server thread/DEBUG]: #29 src/pocketmine/Server(2279): pocketmine\snooze\SleeperHandler->sleepUntil(double 1528992242.5984)
2018-06-14 [18:04:02] [Server thread/DEBUG]: #30 src/pocketmine/Server(2150): pocketmine\Server->tickProcessor()
2018-06-14 [18:04:02] [Server thread/DEBUG]: #31 src/pocketmine/Server(1733): pocketmine\Server->start()
2018-06-14 [18:04:02] [Server thread/DEBUG]: #32 src/pocketmine/PocketMine(290): pocketmine\Server->__construct(BaseClassLoader object, pocketmine\utils\MainLogger object, string /home/gsp_1485/g3.hmserv.eu-19144/, string /home/gsp_1485/g3.hmserv.eu-19144/plugins/)
2018-06-14 [18:04:02] [Server thread/DEBUG]: #33 /home/gsp_1485/g3.hmserv.eu-19144/PocketMine-MP.phar(1): require(string phar:///home/gsp_1485/g3.hmserv.eu-19144/PocketMine-MP.phar/src/pocketmine/PocketMine.php)
```

**Comment by JackMD** [comment]
Fixed in  951c7cb
